pkg/loki/loki.go

```diff
@@ -259,6 +264,18 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+func ValidateConfigTimeouts(c *Config) error {
+	if c.Server.HTTPServerReadTimeout < time.Duration(c.LimitsConfig.QueryTimeout) {
+		return fmt.Errorf("invalid read timeout, the server read timeout (%s) must be larger than the limits timeout (%s)", c.Server.HTTPServerReadTimeout, c.LimitsConfig.QueryTimeout)
+	}
+
+	if c.Server.HTTPServerWriteTimeout < time.Duration(c.LimitsConfig.QueryTimeout) {
```

Comment:
not sure the write timeout has to be checked here. it shouldn't be affected by the query-timeout.